mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubUtils.java

```diff
@@ -0,0 +1,99 @@
+import org.folio.rest.jaxrs.model.PublisherDescriptor;
+import org.folio.rest.jaxrs.model.SubscriberDescriptor;
+import org.folio.rest.tools.PomReader;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+/**
+ * Util class for reading module messaging descriptor
+ */
+public class PubSubUtils {
+
+  public static final String MESSAGING_CONFIG_PATH_PROPERTY = "messaging_config_path";
+  private static final String MESSAGING_CONFIG_FILE_NAME = "MessagingDescriptor.json";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PubSubUtils.class);
+
+  private PubSubUtils() {
+  }
+
+  /**
+   * Reads messaging descriptor file 'MessagingDescriptor.json' and returns {@link DescriptorHolder} that contains
+   * descriptors for module registration as publisher and subscriber.
+   * At first, messaging descriptor is searched in directory by path specified in 'messaging_config_path' system property,
+   * if file was not found then it is searched in classpath.
+   * Location for descriptor directory can be specified as absolute or relative path.
+   * If descriptor directory path is relative then file is searched relative to classpath.
+   * Throws {@link MessagingDescriptorNotFoundException} when messaging descriptor file was not found.
+   *
+   * @return {@link DescriptorHolder}
+   * @throws MessagingDescriptorNotFoundException if messaging descriptor file was not found
+   * @throws IOException if a low-level I/O problem (unexpected end-of-input) occurs while readind file
+   * @throws IllegalArgumentException if parsing file problems occurs (file contains invalid json structure)
+   */
+  public static DescriptorHolder readMessagingDescriptor() throws IOException {
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      File messagingDescriptorFile = getMessagingDescriptorFile();
+      MessagingDescriptor messagingDescriptor = objectMapper.readValue(messagingDescriptorFile, MessagingDescriptor.class);
+
+      return new DescriptorHolder()
+        .withPublisherDescriptor(new PublisherDescriptor()
+            .withModuleName(PomReader.INSTANCE.getModuleName())
+            .withEventDescriptors(messagingDescriptor.getPublications()))
+        .withSubscriberDescriptor(new SubscriberDescriptor()
+          .withModuleName(PomReader.INSTANCE.getModuleName())
+          .withSubscriptionDefinitions(messagingDescriptor.getSubscriptions()));
+    } catch (JsonParseException | JsonMappingException e) {
+      String errorMessage = "Can not read messaging descriptor, cause: " + e.getMessage();
+      LOGGER.error(errorMessage);
+      throw new IllegalArgumentException(e);
+    }
+  }
+
+  private static File getMessagingDescriptorFile() throws MessagingDescriptorNotFoundException {
+    return Optional.ofNullable(System.getProperty(MESSAGING_CONFIG_PATH_PROPERTY))
+      .flatMap(PubSubUtils::getFileByParentPath)
```

Comment:
It can be just 
`      .map(PubSubUtils::getFileByParentPath)
`
instead of 
`      .flatMap(PubSubUtils::getFileByParentPath)
      .map(Optional::of)
`

Reply:
Thank you for review, 
at first it was like you suggest but in case when file was not found 
`getFileByParentPath()` returns empty Optional<File> then `.map(PubSubUtils::getFileByParentPath)` returns `Optional<Optional<File>>` that has empty `Optional<File>`
and then `.orElseGet(() -> getFileFromClassPath(MESSAGING_CONFIG_FILE_NAME))` returns empty `Optional<File>` on which `.orElseThrow()` throws exception cause optional empty,
instead of run `getFileFromClassPath(MESSAGING_CONFIG_FILE_NAME)`.